src/adapters/handlers/prices.ts

```diff
@@ -71,7 +69,18 @@ export function createPricesHandler(
           network,
         }),
       }),
-      responseHeaders
+      {
+        'Cache-Control': `public,max-age=${MAX_AGE},s-maxage=${MAX_AGE}`,
+      },
+      (data: any) => {
+        const dataString = JSON.stringify(data)
+        return {
+          ETag: etag(dataString),
+          'Last-Modified': new Date().toUTCString(),
+          'Content-Type': 'application/json',
```

Comment:
This can go on the normal headers, not the extra as it does not depend on the data